src/GraphQL/Resolvers/MethodResolver.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using GraphQL.Types;
+
+namespace GraphQL.Resolvers
+{
+    /// <summary>
+    /// Experimental. Probably should be a part of <see cref="NameFieldResolver"/>
+    /// It has <see cref="CreateQueryArguments"/> for field registration
+    /// </summary>
+    internal class MethodResolver : IFieldResolver
+    {
+        public static readonly ConcurrentDictionary<MethodInfo, Func<object, object[], object>> cache = new ConcurrentDictionary<MethodInfo, Func<object, object[], object>>();
```

Comment:
public?

Reply:
Okay. I was not sure about the name and whether it should be separated from the NameFieldResolver.

Reply:
> Okay. I was not sure about the name and whether it should be separated from the NameFieldResolver.

Or is it about cache? Fixed public #cache.